source/exe/main.cc

```diff
@@ -1,70 +1,34 @@
 #include <iostream>
 #include <memory>
 
-#include "common/common/compiler_requirements.h"
-#include "common/event/libevent.h"
-#include "common/local_info/local_info_impl.h"
-#include "common/network/utility.h"
-#include "common/stats/stats_impl.h"
-#include "common/stats/thread_local_store.h"
-
 #include "exe/hot_restart.h"
+#include "exe/main_common.h"
 
 #ifdef ENVOY_HANDLE_SIGNALS
 #include "exe/signal_action.h"
 #endif
 
-#include "server/config_validation/server.h"
-#include "server/drain_manager_impl.h"
 #include "server/options_impl.h"
-#include "server/server.h"
-#include "server/test_hooks.h"
 
-#include "ares.h"
 #include "spdlog/spdlog.h"
 
-namespace Envoy {
-namespace Server {
-
-class ProdComponentFactory : public ComponentFactory {
-public:
-  // Server::DrainManagerFactory
-  DrainManagerPtr createDrainManager(Instance& server) override {
-    return DrainManagerPtr{new DrainManagerImpl(server)};
-  }
-
-  Runtime::LoaderPtr createRuntime(Server::Instance& server,
-                                   Server::Configuration::Initial& config) override {
-    return Server::InstanceUtil::createRuntime(server, config);
-  }
-};
-
-} // Server
-} // Envoy
+// NOLINT(namespace-envoy)
```

Comment:
A quirk of the old main.cc was that there was a "namespace Envoy {" despite main() itself not being inside it. This confused the linter into not requiring NOLINT for the old main.cc.